internal/provider/nordvpn/api/getservers.go

```diff
@@ -0,0 +1,80 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/qdm12/gluetun/internal/constants/vpn"
+	"github.com/qdm12/gluetun/internal/models"
+)
+
+var (
+	ErrHTTPStatusCodeNotOK = errors.New("HTTP status code not OK")
+)
+
+func (api *API) GetServers(recommended bool) (servers []models.Server, err error) {
+	var url string
+	if recommended {
+		url = baseURL + "servers/recommendations?limit=10000"
+	} else {
+		url = baseURL + "servers?limit=10000"
+	}
+
+	req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res, err := api.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("%w: %s", ErrHTTPStatusCodeNotOK, res.Status)
+	}
+
+	return parse(res.Body)
+}
+
+func parse(rdr io.Reader) (servers []models.Server, err error) {
+	var jsonServers []server
+	decoder := json.NewDecoder(rdr)
+	if err := decoder.Decode(&jsonServers); err != nil {
+		return nil, fmt.Errorf("failed unmarshaling resp body: %w", err)
```

Comment:
- removed `failed` otherwise it creates repetition on deeply wrapped errors (i.e. `failed to do X: failed to do Y: actual failure`)
- changed `unmarshaling resp body` to `decoding json response` so it's easier to understand in logs (`decoding` is already niche for common people, but `unmarshaling` is even further niche)

```suggestion
		return nil, fmt.Errorf("decoding json response: %w", err)
```